src/states/node.rs

```diff
@@ -516,6 +524,95 @@ impl Node {
         Ok(())
     }
 
+    fn get_section(&self, prefix: &Prefix<XorName>) -> Result<BTreeSet<XorName>, RoutingError> {
+        let section = self.peer_mgr
+            .routing_table()
+            .get_section(&prefix.lower_bound())
+            .ok_or(RoutingError::InvalidSource)?
+            .iter()
+            .cloned()
+            .collect();
+        Ok(section)
+    }
+
+    fn get_section_list(&self, prefix: &Prefix<XorName>) -> Result<GroupList, RoutingError> {
+        Ok(GroupList {
+            pub_ids: self.peer_mgr
+                .get_pub_ids(&self.get_section(prefix)?.into_iter().collect())
```

Comment:
If you inline `get_section` you won't have to. If you update `get_pub_ids` to return a `BTreeSet` (which I think will simplify more code than it will complicate), you can avoid another round of iter/collect conversion.